### PR TITLE
Add Multiple firmware update capsule image support

### PR DESCRIPTION
**Pull request by raghavag**
This patch will add support for updating multiple firmwares
using a single capsule image.

Following modifications are made for existing firmware update flow

1) Gather and validate capsule image
2) State Machine will be set to capsule processing state.
3) Signature of the capsule image is now stored in reserved region
   During each reboot until the end of firmware update, stored signature
   will be compared against the capsule image signature to make sure
   capsule image is not modified until the end of firmware update.
4) Process Capsule image to gather firmware images
5) Each Firmware image information will be stored in reserved region
   using FW_UPDATE_COMP_STATUS. Update pending will be marked to
   update pending state indicating that this image is not processed.
6) Firmware update will use the reserved region comp structures starting
   with the first image with update pending state, update the comp update
   pending field to processing and applies the image. After the update, pending
   field will be updated to Done and updates the status of the update in the
   component structure and moves on to next image found.
7) After all the component structure in the reserved region updating pending
   field are set to Done. Firmware update mode is exited.

Signed-off-by: Raghava Gudla <raghava.gudla@intel.com>

### CODE REVIEW

File: BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h [hunk 1]
```diff
@@ -20,29 +20,38 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
...
+#define FW_UPDATE_PARTITION_A   0
+#define FW_UPDATE_PARTITION_B   1
+
+#define FW_UPDATE_SIG_LENGTH    256
 
-#define FW_UPDATE_PARTITION_A 0
-#define FW_UPDATE_PARTITION_B 1
+#define RTC_PORT_ID             0xC3
+#define PCH_LPC_REG_P2SB_UNHIDE 0xE1
+#define PCH_LPC_REG_P2SB_BAR    0x10
```
Comment (gdong1) [inline BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h]: Line 44 to 46  is not related with firmware update, should be removed from this header file.

File: BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h [hunk 2]
```diff
@@ -61,8 +70,8 @@ typedef enum {
 } BOOT_PARTITION;
 
 #define ESRT_FIRMWARE_RESOURCE_VERSION    0x1
-#define ESRT_FIRMWARE_RESOURCE_COUNT      0x1
-#define ESRT_FIRMWARE_RESOURCE_COUNT_MAX  0x1
+#define ESRT_FIRMWARE_RESOURCE_COUNT      0x3
```
Comment (gdong1) [inline BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h]: why define ESRT_FIRMWARE_RESOURCE_COUNT, is there anyone using it?

Comment (raghavag) [reply BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h]: Removed them, Using MAX_FW_COMPONENTS instead

File: BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h (hunk 1 above)
Comment (raghavag) [reply BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h]: Removed them

File: BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c [hunk 3]
```diff
@@ -423,27 +420,13 @@ UpdateFwst (
...
-
-  CopyMem(&FwstAcpiTablePtr->EsrtTableEntry.FwClass, &gEsrtSystemFirmwareGuid, sizeof (EFI_GUID)); 
-  FwstAcpiTablePtr->EsrtTableEntry.FwType = ESRT_FW_TYPE_SYSTEMFIRMWARE;
-
-  //
-  // Get current version and lowest supported SVN version
-  //
-  if (LdrGlobal->VerInfoPtr != NULL) {
-    Version = LdrGlobal->VerInfoPtr;
-    FwstAcpiTablePtr->EsrtTableEntry.FwVersion = Version->ImageVersion.SecureVerNum;
```
Comment (gdong1) [inline BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c]: where FwstAcpiTablePtr->EsrtTableEntry.FwVersion is set?

Comment (raghavag) [reply BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c]: This is a to do part for me. Will update in the next version

Comment (gdong1) [reply BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c]: it is ok for me. please create an issue to track it.

Comment (raghavag) [reply BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c]: Done Created one for this

**Pull request merged**